apps/dashboard/app/javascript/packs/script_edit.js

```diff
@@ -121,10 +121,33 @@ function addInProgressField(event) {
   justAdded.find('[data-select-toggler]')
            .on('click', (event) => { enableOrDisableSelectOption(event) });
 
+  justAdded.find('[data-fixed-toggler]')
+           .on('click', (event) => { toggleFixed(event) });
+
   const entireDiv = event.target.parentElement.parentElement.parentElement;
   entireDiv.remove();
 }
 
+function fixedFieldEnabled(checkbox, dataElement) {
+  dataElement.disabled = true;
+  const input = $('<input>').attr('type','hidden').attr('name', dataElement.name).attr('value', dataElement.value);
+  $(checkbox).after(input);
```

Comment:
Do we need to change the HTML tag type or can we just toggle the `readonly` attribute?

Reply:
Select elements do not have readonly state, so to get the current behaviour we need to disable and add hidden field.

> Do we need to change the HTML tag 
Not sure if this means to change from `select` to `input type=text` to make it `readonly`, but it seems a more complex solution

Reply:
> Select elements do not have readonly state, so to get the current behaviour we need to disable and add hidden field.

Editing the page on FF seems to respect the `readonly` - though MDN docs say it shouldn't. Further reading suggests it was deprecated in html5? But it's unclear how to find official RFC for the same.

![image](https://github.com/OSC/ondemand/assets/4874123/1a4e6ead-08bf-4af2-bdc5-8783890c078c)

> Not sure if this means to change from select to input type=text to make it readonly, but it seems a more complex solution

I thought that's what this was doing - but now see it adds a hidden element instead of replacing the original.  I guess this is because the browser won't send a disabled form element? Seems like it'll send `readonly` - but I cannot actually find if it's actually been deprecated on select fields or what.

Reply:
OK - I think what you have is what we should go with. That is (if I'm reading it right),
* disabling the original element
* creating a new hidden element with the same name so that it'll be passed in the form

and of course the inverse too.

Maybe there's a way to clean this up so it's a little more apparent what's going on? It's created in one place and removed in another. Maybe just colocate these?

Reply:
> OK - I think what you have is what we should go with

I think this because I cannot find any docs on why my example shown works. The currrent RFC (below) for html5 doesn't specify it and MDN docs (also below) explicitly call it out as not working.

Which is to say - relying on adding `readonly` to a select widget is likely unstable (not to mention how other browsers react, I only looked doing this on firefox).

https://www.w3.org/TR/html5-author/the-select-element.html#the-select-element
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly

Reply:
Yes, it is disabling and adding a hidden field with the same name as the disabled field will not be sent in the request.
I tried the readonly first as it was more intuitive, but in my Firefox did not work. Reading later I came across that select elements do not have a readonly and I found the recommended way of adding the hidden field when disabled.

I will colocate the methods and add comments to make it more obvious what is happening.

Reply:
I added some comments. The add/remove code is close to each other. Not in the same method to reuse the add in the initialization code.